### PR TITLE
[FIX] account_move_line : fix distinct on partners

### PR DESCRIPTION
**Pull request by BHanq**
Whenever invoices are imported in batch, the account_id are not taking the default property_account of the partners but the default one of the company. This appears because, the partners are distinct on the company and not on the partners. Which leads to add only one partner in the lead for each company, even if there are multiple related partners.

Related to opw-3284713

Description of the issue/feature this PR addresses: Whenever invoices are imported in batch, the account_id are not taking the default property_account of the partners but the default one of the company.

Current behavior before PR: The account is right for only one partner and the invoice of the rest of the partners are created with the account of the company.

Desired behavior after PR is merged: Compute the right account for each partner while creating invoice in batch (or import).

### CODE REVIEW

File: addons/account/models/account_move_line.py [hunk 1]
```diff
@@ -501,7 +501,7 @@ def _compute_account_id(self):
                        AND line.id != ANY(%(current_ids)s)
                 ),
                 properties AS(
-                    SELECT DISTINCT ON (property.company_id, property.name)
+                    SELECT DISTINCT ON (property.res_id, property.name)
```
Comment (jco-odoo) [inline addons/account/models/account_move_line.py:504]: Probably wrong, because you can different property's for the same res_id for different companies.

Comment (BHanq) [reply addons/account/models/account_move_line.py:504]: Would adding `SELECT DISTINCT ON (property.company_id, property.res_id, property.name)` be better ?

File: addons/account/models/account_move_line.py [hunk 2]
```diff
@@ -494,27 +494,29 @@ def _compute_account_id(self):
                            'account.move' AS model,
                            line.move_id AS id,
                            NULL AS account_type,
+                           NULL::integer AS company_id,
```
Comment (jco-odoo) [inline addons/account/models/account_move_line.py:497]: Why do we need to cast NULL?

**Comment by jco-odoo** [comment]
I wonder if we should not have a test for it as well.  Somehow in the standard flow, this problem does not occur, only when you import. Otherwise, this lgtm.  
 @smetl Could you check further?

**Comment by smetl** [comment]
@BHanq Plz fix the commit message
`[FIX] account: Fix default account_id on invoice terms lines`

**Comment by BHanq** [comment]
> @BHanq Plz fix the commit message `[FIX] account: Fix default account_id on invoice terms lines`

Yes sure ! I'll first add a test
Thanks for the feedback.

File: addons/account/models/account_move_line.py [hunk 3]
```diff
@@ -537,6 +540,7 @@ def _compute_account_id(self):
                            'res.company' AS model,
                            account.company_id AS id,
                            account.account_type AS account_type,
+                           NULL::integer AS company_id,
```
[the remark this answers is not in the document]
Comment (BHanq) [reply addons/account/models/account_move_line.py:543]: Yes but as it is an union, each query must have the exact same number of columns and type in each column

File: addons/account/models/account_move_line.py (hunk 2 above)
Comment (BHanq) [reply addons/account/models/account_move_line.py:497]: image

With the Union ALL, each column must have the same type between queries, NULL is taken as string I guess

**Comment by BHanq** [comment]
Similar to https://github.com/odoo/odoo/commit/c66d9ee2a160c3bad745b1394308c3e48f5f2e91

**Pull request closed without merging**